Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -756,18 +747,45 @@ eval RetryTimeUpdate state = do
   (updateAndExit state { data = state.data { locationLastUpdatedTime = "" }, props = state.props {refreshAnimation = true}} $ Refresh state { data = state.data { locationLastUpdatedTime = "" }, props = state.props {refreshAnimation = true}})
 
 eval (TimeUpdate time lat lng) state = do
-  let isDriverNearBy = ((getDistanceBwCordinates (getLastKnownLocValue ST.LATITUDE lat) (getLastKnownLocValue ST.LONGITUDE lng)  state.data.activeRide.src_lat state.data.activeRide.src_lon) < 0.05)
-      newState = state { data = state.data { activeRide{isDriverArrived = if ((state.props.currentStage == ST.RideAccepted || state.props.currentStage == ST.ChatWithCustomer) && not state.data.activeRide.notifiedCustomer) then isDriverNearBy else state.data.activeRide.isDriverArrived},currentDriverLat= getLastKnownLocValue ST.LATITUDE lat,  currentDriverLon = getLastKnownLocValue ST.LONGITUDE lng, locationLastUpdatedTime = (convertUTCtoISC time "hh:mm a") }}
-  _ <- pure $ setValueToLocalStore IS_DRIVER_AT_PICKUP (show (newState.data.activeRide.isDriverArrived || newState.data.activeRide.notifiedCustomer))
-  _ <- pure $ setValueToLocalStore LOCATION_UPDATE_TIME (convertUTCtoISC time "hh:mm a")
+  let newState = state { data = state.data { currentDriverLat= getLastKnownLocValue ST.LATITUDE lat,  currentDriverLon = getLastKnownLocValue ST.LONGITUDE lng, locationLastUpdatedTime = (convertUTCtoISC time "hh:mm a") }}
+  void $ pure $ setValueToLocalStore IS_DRIVER_AT_PICKUP (show newState.data.activeRide.notifiedCustomer)
+  void $ pure $ setValueToLocalStore LOCATION_UPDATE_TIME (convertUTCtoISC time "hh:mm a")
   continueWithCmd newState [ do
-    _ <- if (getValueToLocalNativeStore IS_RIDE_ACTIVE == "false") then checkPermissionAndUpdateDriverMarker newState else pure unit
-    pure AfterRender
+    void $ if (getValueToLocalNativeStore IS_RIDE_ACTIVE == "false") then checkPermissionAndUpdateDriverMarker newState else pure unit
+    case state.data.config.waitTimeConfig.enableWaitTime, state.props.currentStage, state.data.activeRide.notifiedCustomer, state.data.snappedOrigin of
+      true, ST.RideAccepted, false, Nothing -> pure (UpdateOriginDist lat lng)
+      true, ST.RideAccepted, false, Just snapped -> do
+        let dist = (getDistanceBwCordinates (getLastKnownLocValue ST.LATITUDE lat) (getLastKnownLocValue ST.LONGITUDE lng)  snapped.lat snapped.lon)
+            isDriverNearBy = ( dist < state.data.config.waitTimeConfig.thresholdDist)
+        pure if isDriverNearBy then NotifyAPI else AfterRender
+      _, _, _, _ -> pure AfterRender
     ]
 
+eval (UpdateOriginDist lat lng) state =
+  continueWithCmd state [ do
+    void $ launchAff $ flowRunner defaultGlobalState $ runExceptT $ runBackT $ do
+      push <- liftFlowBT $ getPushFn Nothing "HomeScreen"
```

Comment:
if Remote.rideRoute is a Flow function we don't need `runExceptT $ runBackT`. remove lift $ lift and change  liftFlowBT to liftFlow